force-app/main/default/classes/LeadTriggerHandlerTest.cls

```diff
@@ -0,0 +1,153 @@
+        new Lead(Company = 'One', LastName = 'Doe6', LeadSource = 'Website', Email = null, Phone = '1234567980', Industry = 'Healthcare'),
+        new Lead(Company = 'One', LastName = 'Doe7', LeadSource = 'Website', Email = null, Phone = '1234567980', Industry = 'Technology'),
+        new Lead(Company = 'One', LastName = 'Doe8', LeadSource = 'Website', Email = null, Phone = null, Industry = 'Technology'),
+        
+        new Lead(Company = 'One', LastName = 'Doe9', LeadSource = null, Email = 'test@example.com', Phone = null, Industry = 'Healthcare'),
+        new Lead(Company = 'One', LastName = 'Doe10', LeadSource = null, Email = 'test@example.com', Phone = '1234567980', Industry = 'Healthcare'),
+        new Lead(Company = 'One', LastName = 'Doe11', LeadSource = null, Email = 'test@example.com', Phone = '1234567980', Industry = 'Technology'),
+        new Lead(Company = 'One', LastName = 'Doe12', LeadSource = null, Email = 'test@example.com', Phone = null, Industry = 'Technology'),
+        
+        new Lead(Company = 'One', LastName = 'Doe13', LeadSource = 'Website', Email = 'test@example.com', Phone = null, Industry = 'Healthcare'),
+        new Lead(Company = 'One', LastName = 'Doe14', LeadSource = 'Website', Email = 'test@example.com', Phone = null, Industry = 'Technology'),
+        new Lead(Company = 'One', LastName = 'Doe15', LeadSource = 'Website', Email = 'test@example.com', Phone = '1234567980', Industry = 'Healthcare'),
+        new Lead(Company = 'One', LastName = 'Doe16', LeadSource = 'Website', Email = 'test@example.com', Phone = '1234567980', Industry = 'Technology')
+        };
+        
+        Test.startTest();
+        LeadTriggerHandler.handleAutoLeadScoring(leads);
+        Test.stopTest();
+        
+        // Query updated leads and validate scoring logic
+            List<Lead> updatedLeads = [SELECT Id, LeadSource, Email, Phone, Industry, Lead_Score__c FROM Lead WHERE Id IN :leads];        
+            // Add validation for expected scores
+            for (Lead lead : updatedLeads) {
+                Integer expectedScore = calculateExpectedScore(lead.LeadSource, lead.Email, lead.Phone, lead.Industry);
+                System.assertEquals(expectedScore, lead.Lead_Score__c, 'Incorrect Lead Score for Lead: ' + lead.LastName);
+            }
+        }                
+        // Helper function to determine the expected Lead Score dynamically
+        public static Integer calculateExpectedScore(String leadSource, String email, String phone, String industry) {
+            Integer score = 0;
+        
+            if (leadSource == 'Website' && email != null) {
+                score += 3;
+            }
+            if (phone != null) {
+                score += 5;
+            }
+            if (industry == 'Technology') {
+                score += 10;
+            }
+        
+            return Math.min(score, 18);
```

Comment:
Awesome! There might also be some instances where the score comes back as (8, 13, 15). Try to add validations for those times.